src/arch_switch.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * This software is licensed under the terms of the MIT-License
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2019, p-sam <p-sam@d3vs.net>.
+ */
+
+#include "arch_switch.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <switch/runtime/devices/socket.h>
+#include <switch/runtime/nxlink.h>
+#include <switch/services/applet.h>
+#include <switch/services/fs.h>
+
+#include "taisei.h"
```

Comment:
This should go before all other includes for two reasons:

* We use precompiled headers by default, and the pch includes this file early, so that's more consistent.
* We have an manual upkeep script that inserts missing copyright notices, include guards, and `#include "taisei.h"`s. It'll get confused if this include is not at the top.

You also don't need any of those standard library includes; `"taisei.h"` already includes them.